zap/src/output/typescript/mod.rs

```diff
@@ -185,18 +190,18 @@ pub trait Output {
 		}
 	}
 
-	fn push_arg_ty(&mut self, ty: &Ty) {
+	fn push_arg_ty(&mut self, ty: &Ty, config: &Config) {
```

Comment:
Let's not make `config` an argument of `push_ty`/`push_arg_ty`. ~Instead add it to the trait itself~ - I realised you can't do this in rust. I'll try to think about alternative options.